### PR TITLE
[OPENY-72] openy_addthis decoupling

### PR DESCRIPTION
**Pull request by hamrant**
## Steps for review

- [x] Login as admin
- [x] go to any blog or news page
- [x] check that you can see social sharing block
- [x] create new landing page with "Social share icons" paragraph
- [x] check that you can see social sharing block
- [x] go to /admin/modules/uninstall
- [x] uninstall "OpenY Demo AddThis" module
- [x] check that you can see "Remove field values: addthis on entity type Content." message with link for "Openy addThis" module
uninstall-addthis
- [x] Click on link "Remove field values"
- [x] Press "Delete all addthis field values" button on form
- [x] After this you will be redirected back to  /admin/modules/uninstall
- [x] check that now you can uninstall "Openy addThis" module
- [x] uninstall "OpenY AddThis" module
- [x] go to /admin/structure/paragraphs_type
- [x] check that  "Social share icons" paragraph not exist in this list
- [x] go to any blog or news page
- [x] check that  "Social share icons" not exist in this pages
- [x] also check created landing page with this paragraph
- [x] go to /admin/modules
- [x] install "OpenY AddThis" and  "OpenY Demo AddThis"  modules (Demo module contain openy_addthis.settings.public_id)
- [x] go to any blog page and resave node - check that you can see  "Social share icons" on blog page
- [x] check that all components related to "OpenY AddThis" module was restored and works fine

### CODE REVIEW

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build548 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build548 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build548 |
| Installation process | http://install.openy.cibox.tools/build548/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/548/

**Review by ddrozdik — APPROVED** [review]

File: modules/custom/openy_addthis/src/Form/PrepareUninstallForm.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    $tables_to_update = [];
+
+    foreach ($this->entityTypeManager->getDefinitions() as $entity_type) {
+      if ($entity_type->id() == 'node') {
+        $tables_to_update[] = $entity_type->getDataTable();
+        $tables_to_update[] = $entity_type->getRevisionDataTable();
+      }
+    }
+
+    $db_connection = Database::getConnection();
```
Comment (podarok) [inline modules/custom/openy_addthis/src/Form/PrepareUninstallForm.php]: could you use Dependency Injection to pass DB connection?

**Review by podarok — CHANGES_REQUESTED** [review]

**Review by podarok — APPROVED** [review]

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build556 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build556 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build556 |
| Installation process | http://install.openy.cibox.tools/build556/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/556/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build556/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build556/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build556/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build556/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build556/blogWCAG2AAhtmlcs.txt

**Pull request merged**